django_netjsonconfig/utils.py

```diff
@@ -132,6 +135,9 @@ def get_controller_urls(views_module):
         url(r'^controller/download-config/(?P<pk>[^/]+)/$',
             views_module.device_download_config,
             name='download_config_legacy'),
+        url(r'^controller/update-info/(?P<pk>[^/]+)/$',
+            views_module.device_update_info,
+            name='update_info_legacy'),
```

Comment:
we can avoid the legacy URL since this is a new URL

Reply:
I am a little bit irritated because here it only works with the legacy URL `^controller/update-info/(?P<pk>[^/]+)/$`. And in `openwisp-config` the URL is also `$BASEURL/update-info/$UUID/` with `$BASEURL` being `$URL/controller`.

What am I missing?

Reply:
you're right, we haven't updated openwisp-config to use the new URLs yet, so please go ahead and put the legacy URL back since it seems we need it right now (the patch in openwisp-config is also using that one right?)

Reply:
Yes, `openwisp-config` uses only URLs like `^controller/<api-endpoint>/` and not URLs like `^controller/device/<api-endpoint>/`.

See:
https://github.com/openwisp/openwisp-config/blob/master/openwisp-config/files/openwisp.agent#L73
https://github.com/openwisp/openwisp-config/blob/master/openwisp-config/files/openwisp.agent#L585

So we will need to keep the legacy URLs for quite some time, otherwise older clients won't be able to connect to an updated server anymore.

I re-added the legacy URL.